xmppserver/src/main/java/org/jivesoftware/database/bugfix/Issue369and370.java

```diff
@@ -29,29 +40,34 @@ public static void executeMigration() {
 
         // For conversations that are not external, assign a unique roomID based on roomJID using information from the ofConversation table
         try (Connection con = DbConnectionManager.getConnection();
-             PreparedStatement pstmt = con.prepareStatement("SELECT room, conversationID FROM ofConversation WHERE isExternal = 0 ORDER BY startDate ASC");
+             PreparedStatement pstmt = con.prepareStatement("SELECT room, conversationID FROM ofConversation WHERE isExternal = 0 ORDER BY startDate ASC", ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
              ResultSet rs = pstmt.executeQuery()) {
 
+            pstmt.setFetchSize(250);
+            pstmt.setFetchDirection(ResultSet.FETCH_FORWARD);
             while (rs.next()) {
                 String roomJID = rs.getString("room");
                 long conversationID = rs.getLong("conversationID");
                 long roomID = roomJIDToRoomIDMap.computeIfAbsent(roomJID, k -> roomIDSequenceManager.nextUniqueID());
                 conversationIDToRoomIDMap.put(conversationID, roomID);
             }
-            Log.info("1 of 7. Generated room IDs for unique rooms");
+
+            pstmt.setFetchSize(250);
```

Comment:
Why do this here? Is this intended or a a copy/paste bug?

Reply:
Yes, they are redundant and misplaced calls. Removed now.